src/containers/my-courses/add-course-with-input/AddCourseWithInput.tsx

```diff
@@ -33,10 +44,23 @@ const AddCourseWithInput = () => {
     }
   })
 
+  const debounceOnChange = useDebounce((text: string) => {
+    searchRef.current = text
+    void fetchData()
+  })
+
   const onChange = (e: ChangeEvent<HTMLInputElement>) => {
-    setInputValue(e.target.value)
+    const value = e.target.value
+    setInputValue(value)
+    debounceOnChange(value)
```

Comment:
Maybe putting "e.target.value" in debounce OnChange is better to avoid creating new variables.

Reply:
> Maybe putting "e.target.value" in debounce OnChange is better to avoid creating new variables.

fixed